### PR TITLE
chore(logstreams): fix sigsegv

### LINKED ISSUE #2324: AssertionError in LockBlockIndex rollback transaction call
In tests I sometimes see the following assertion error, which indicates that we try to rollback a transaction which does not own the rocksdb native handle. 

```
Exception in thread "0.0.0.0:25636-zb-actors-0" java.lang.AssertionError
	at org.rocksdb.Transaction.rollback(Transaction.java:215)
	at io.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.rollbackInternal(ZeebeTransaction.java:107)
	at io.zeebe.db.impl.rocksdb.transaction.DefaultDbContext.runInNewTransaction(DefaultDbContext.java:174)
	at io.zeebe.db.impl.rocksdb.transaction.DefaultDbContext.runInTransaction(DefaultDbContext.java:143)
	at io.zeebe.db.impl.rocksdb.transaction.ZeebeTransactionDb.ensureInOpenTransaction(ZeebeTransactionDb.java:154)
	at io.zeebe.db.impl.rocksdb.transaction.ZeebeTransactionDb.whileTrue(ZeebeTransactionDb.java:266)
	at io.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.whileTrue(TransactionalColumnFamily.java:117)
	at io.zeebe.logstreams.impl.log.index.LogBlockIndex.lookupBlockPosition(LogBlockIndex.java:114)
	at io.zeebe.logstreams.impl.log.index.LogBlockIndex.lookupBlockAddress(LogBlockIndex.java:88)
	at io.zeebe.logstreams.log.BufferedLogStreamReader.lookUpBlockAddressForPosition(BufferedLogStreamReader.java:327)
	at io.zeebe.logstreams.log.BufferedLogStreamReader.seek(BufferedLogStreamReader.java:111)
	at io.zeebe.logstreams.log.BufferedLogStreamReader.seekToFirstEvent(BufferedLogStreamReader.java:126)
	at io.zeebe.logstreams.processor.StreamProcessorController.recoverFromSnapshot(StreamProcessorController.java:167)
	at io.zeebe.logstreams.processor.StreamProcessorController.onActorStarting(StreamProcessorController.java:110)
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:87)
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53)
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:189)
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154)
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135)
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112)
```

### LINKED ISSUE #2369: SIGSEGV on broker shut down
Currently we observe in the qa and broker core tests quite often, an segmentation fault on closing the broker. The problem seems that the services are stopped, for example the log block index (which closes his rocks db instance), but the stream processor is still running. The processor will try to find the next event and uses the buffered reader, which internally uses the block index. 

The index will try to access the rocks db, even if it is already closed. In the happy path we only get a NPE exception or Assertion error, because of an check in the rocks db java impl https://github.com/zeebe-io/zeebe/issues/2324. In the worst case we get an segmentation fault

```java
Stack: [0x00007f8120881000,0x00007f8120982000],  sp=0x00007f812097fe50,  free space=1019k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [librocksdbjni2726119066446171792.so+0x622503]  rocksdb::TransactionBaseImpl::GetIterator(rocksdb::ReadOptions const&, rocksdb::ColumnFamilyHandle*)+0xa3
j  org.rocksdb.Transaction.getIterator(JJJ)J+0
j  org.rocksdb.Transaction.getIterator(Lorg/rocksdb/ReadOptions;Lorg/rocksdb/ColumnFamilyHandle;)Lorg/rocksdb/RocksIterator;+42
j  io.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.newIterator(Lorg/rocksdb/ReadOptions;Lorg/rocksdb/ColumnFamilyHandle;)Lorg/rocksdb/RocksIterator;+6
j  io.zeebe.db.impl.rocksdb.transaction.DefaultDbContext.newIterator(Lorg/rocksdb/ReadOptions;Lorg/rocksdb/ColumnFamilyHandle;)Lorg/rocksdb/RocksIterator;+6
j  io.zeebe.db.impl.rocksdb.transaction.ZeebeTransactionDb.newIterator(JLio/zeebe/db/DbContext;Lorg/rocksdb/ReadOptions;)Lorg/rocksdb/RocksIterator;+18
j  io.zeebe.db.impl.rocksdb.transaction.ZeebeTransactionDb.lambda$whileTrue$8(JLio/zeebe/db/DbContext;Lio/zeebe/db/DbKey;Lio/zeebe/db/DbValue;Lio/zeebe/db/KeyValuePairVisitor;Lio/zeebe/db/impl/rocksdb/transaction/ZeebeTransaction;)V+7
j  io.zeebe.db.impl.rocksdb.transaction.ZeebeTransactionDb$$Lambda$550.run(Lio/zeebe/db/impl/rocksdb/transaction/ZeebeTransaction;)V+25
J 5109 C1 io.zeebe.db.impl.rocksdb.transaction.ZeebeTransactionDb$$Lambda$551.run()V (12 bytes) @ 0x00007f8131c91d64 [0x00007f8131c91ae0+0x284]
j  io.zeebe.db.impl.rocksdb.transaction.DefaultDbContext.runInNewTransaction(Lio/zeebe/db/TransactionOperation;)V+8
j  io.zeebe.db.impl.rocksdb.transaction.DefaultDbContext.runInTransaction(Lio/zeebe/db/TransactionOperation;)V+21
J 5107 C1 io.zeebe.db.impl.rocksdb.transaction.ZeebeTransactionDb.ensureInOpenTransaction(Lio/zeebe/db/DbContext;Lio/zeebe/db/impl/rocksdb/transaction/ZeebeTransactionDb$TransactionConsumer;)V (14 bytes) @ 0x00007f8131c91434 [0x00007f8131c90fc0+0x474]
j  io.zeebe.db.impl.rocksdb.transaction.ZeebeTransactionDb.whileTrue(JLio/zeebe/db/DbContext;Lio/zeebe/db/DbKey;Lio/zeebe/db/DbValue;Lio/zeebe/db/KeyValuePairVisitor;)V+16
j  io.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.whileTrue(Lio/zeebe/db/DbContext;Lio/zeebe/db/KeyValuePairVisitor;Lio/zeebe/db/DbKey;Lio/zeebe/db/DbValue;)V+13
j  io.zeebe.logstreams.impl.log.index.LogBlockIndex.lookupBlockPosition(Lio/zeebe/logstreams/impl/log/index/LogBlockIndexContext;J)J+36
j  io.zeebe.logstreams.impl.log.index.LogBlockIndex.lookupBlockAddress(Lio/zeebe/logstreams/impl/log/index/LogBlockIndexContext;J)J+3
j  io.zeebe.logstreams.log.BufferedLogStreamReader.lookUpBlockAddressForPosition(J)J+9
j  io.zeebe.logstreams.log.BufferedLogStreamReader.seek(J)Z+26
j  io.zeebe.logstreams.log.BufferedLogStreamReader.seekToFirstEvent()V+4
j  io.zeebe.logstreams.processor.StreamProcessorController.recoverFromSnapshot()J+63
j  io.zeebe.logstreams.processor.StreamProcessorController.onActorStarting()V+87
J 3691 C1 io.zeebe.util.sched.ActorTask$$Lambda$54.run()V (8 bytes) @ 0x00007f813195e904 [0x00007f813195e800+0x104]
J 3814 C2 io.zeebe.util.sched.ActorThread.doWork()V (103 bytes) @ 0x00007f81319d3c58 [0x00007f81319d3740+0x518]
J 3673 C2 io.zeebe.util.sched.ActorThread.run()V (56 bytes) @ 0x00007f8131952c2c [0x00007f8131952ba0+0x8c]
```

[hs_err_pid7233.log](https://github.com/zeebe-io/zeebe/files/3105993/hs_err_pid7233.log)

### PR DESCRIPTION
**Pull request by ChrisKujawa**
### TL;DR:

  * reduce starting phase time on StreamProcessorController
   this causes problems on closing the service/actor

----

Problem was that the starting phase (`#onActorStarting`) from the `StreamProcessorController` takes quite a bit of time and the `StreamProcessorService#start` uses `startContext#async` to start the controller.
This means the `StreamProcessorService` is successfully installed when the `#onActorStarting` of the `StreamProcessorController` is completed. 

Some tests have a short execution time, which means the `StreamProcessorService` is not completely installed  when `broker#close` is called. This also means that the `#stop` method is not called of the `StreamProcessorService`, since the installation is not yet done. So it will access resources like the LogBlockIndex even if they already closed.

One way to solve this is to mark the service interruptable, but this solves only one problem. The next problem is that we can't interrupt actor jobs, which means if the actor is submitted it will execute `actor#onActorStarting`, even if the `actor#close` is called in the meantime. The actor can only interrupted between the phase callbacks. That is the reason why I moved the snapshot recovery into the `#onActorStarted` method. There is also no real reason why it was in the `#onActorStarting` callback. 

**So keep in mind**: `onActorStarting` only for setup code like assign fields etc. 

closes #2369 
closes #2324

### CODE REVIEW

**Review by saig0 — APPROVED** [review]
bors r+
